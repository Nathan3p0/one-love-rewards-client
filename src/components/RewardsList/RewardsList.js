import React from 'react'
import './RewardsList.css'

const RewardsList = (props) => {
    return (
        <>
            <h3>Current Rewards Available:</h3>
            <ul>
                <li>
                    <div className="user__content-rewards--item">
                        <div className="user__content-rewards--points">
                            <p>100pts</p>
                        </div>
                        <div className="user__content-rewards--reward">
                            <p>$5 off your purchase</p>
                        </div>
                    </div>
                </li>
                <li>
                    <div className="user__content-rewards--item">
                        <div className="user__content-rewards--points">
                            <p>250pts</p>
                        </div>
                        <div className="user__content-rewards--reward">
                            <p>$10 off your purchase</p>
                        </div>
                    </div>
                </li>
                <li>
                    <div className="user__content-rewards--item">
                        <div className="user__content-rewards--points">
                            <p>500pts</p>
                        </div>
                        <div className="user__content-rewards--reward">
                            <p>$25 off your purchase</p>
                        </div>
                    </div>
                </li>
                <li>
                    <div className="user__content-rewards--item">
                        <div className="user__content-rewards--points">
                            <p>1000pts</p>
                        </div>
                        <div className="user__content-rewards--reward">
                            <p>$50 off your purchase</p>
                        </div>
                    </div>
                </li>
            </ul>
        </>
    )
}

export default RewardsList;