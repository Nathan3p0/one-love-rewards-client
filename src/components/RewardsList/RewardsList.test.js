import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import RewardsList from './RewardsList'

describe('Rewards List Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<RewardsList />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<RewardsList />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<RewardsList
            pointsRequired={150}
            points={225}
            discount={'$5 off'}
            button={true}
        />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})