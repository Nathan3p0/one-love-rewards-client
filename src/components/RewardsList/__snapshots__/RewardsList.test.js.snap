// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rewards List Component Rendering Tests renders to UI as expected 1`] = `
Array [
  <h3>
    Current Rewards Available:
  </h3>,
  <ul />,
]
`;

exports[`Rewards List Component Rendering Tests renders to UI as expected with PROPS 1`] = `
Array [
  <h3>
    Current Rewards Available:
  </h3>,
  <ul />,
]
`;
